pyro/poutine/trace.py

```diff
@@ -82,10 +80,7 @@ def log_pdf(self, site_filter=lambda name, site: True):
                     site_log_p = site["log_pdf"]
                 except KeyError:
                     args, kwargs = site["args"], site["kwargs"]
-                    if isinstance(site["fn"], TorchDistribution):
-                        site_log_p = site["fn"].log_prob(site["value"]).sum() * site["scale"]
-                    else:
-                        site_log_p = site["fn"].log_prob(site["value"], *args, **kwargs).sum() * site["scale"]
+                    site_log_p = site["fn"].log_prob(site["value"], *args, **kwargs).sum() * site["scale"]
```

Comment:
@eb8680 Now that `sample_shape` is handled by `Reshape`, we no longer need this `isinstance` kludge.